app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt

```diff
@@ -51,11 +48,18 @@ class HomePresenter(
     private val tokenKeyProvider: TokenKeyProvider,
     private val homeElementItemMapper: HomeElementItemMapper,
     private val resourcesProvider: ResourcesProvider,
-    private val newBuyFeatureToggle: NewBuyFeatureToggle,
-    private val accountStorageContract: UserSignUpDetailsStorage,
-    private val authInteractor: AuthInteractor
+    private val newBuyFeatureToggle: NewBuyFeatureToggle
 ) : BasePresenter<HomeContract.View>(), HomeContract.Presenter {
 
+    private lateinit var fallBackUsdcTokenForBuy: Token
+    private var nickName: Username? = null
+
+    init {
+        launch {
+            fallBackUsdcTokenForBuy = userInteractor.getTokensForBuy(listOf(USDC_SYMBOL)).first()
```

Comment:
fallBack -> fallback

and i suggest you put this preloading on `onAttach` method and wrapping it in try catch